Tests/ApolloSQLiteTests/CachePersistenceTests.swift

```diff
@@ -160,3 +114,60 @@ class CachePersistenceTests: XCTestCase {
     }
   }
 }
+
+extension CachePersistenceTests {
+  private func runTestFetchAndPersist(
+    againstFileAt sqliteFileURL: URL,
+    file: StaticString = #file,
+    line: UInt = #line
+  ) {
+      let query = HeroNameQuery()
+
+      SQLiteTestCacheProvider.withCache(fileURL: sqliteFileURL) { (cache) in
+        let store = ApolloStore(cache: cache)
+        let networkTransport = MockNetworkTransport(body: [
+          "data": [
+            "hero": [
+              "name": "Luke Skywalker",
+              "__typename": "Human"
+            ]
+          ]
+        ], store: store)
+        let client = ApolloClient(networkTransport: networkTransport, store: store)
+
+        let networkExpectation = self.expectation(description: "Fetching query from network")
+        let newCacheExpectation = self.expectation(description: "Fetch query from new cache")
+
+        client.fetch(query: query, cachePolicy: .fetchIgnoringCacheData) { outerResult in
+          defer { networkExpectation.fulfill() }
+
+          switch outerResult {
+          case .failure(let error):
+            XCTFail("Unexpected error: \(error)", file: file, line: line)
+            return
+          case .success(let graphQLResult):
+            XCTAssertEqual(graphQLResult.data?.hero?.name, "Luke Skywalker", file: file, line: line)
+            // Do another fetch from cache to ensure that data is cached before creating new cache
+            client.fetch(query: query, cachePolicy: .returnCacheDataDontFetch) { innerResult in
+              SQLiteTestCacheProvider.withCache(fileURL: sqliteFileURL) { cache in
+                let newStore = ApolloStore(cache: cache)
+                let newClient = ApolloClient(networkTransport: networkTransport, store: newStore)
+                newClient.fetch(query: query, cachePolicy: .returnCacheDataDontFetch) { newClientResult in
+                  defer { newCacheExpectation.fulfill() }
+                  switch newClientResult {
+                  case .success(let newClientGraphQLResult):
+                    XCTAssertEqual(newClientGraphQLResult.data?.hero?.name, "Luke Skywalker", file: file, line: line)
+                  case .failure(let error):
+                    XCTFail("Unexpected error with new client: \(error)", file: file, line: line)
+                  }
+                  _ = newClient // Workaround for a bug - ensure that newClient is retained until this block is run
```

Comment:
I don't necessarily think this is a bug - but I would leave the note about ensuring it's retained until the block is run.

Reply:
This was existing code that I just moved into a shared function so that I could pass the parameter but also make sure that the unit test is picked up by the XCTest sweep.

Do you still want me to update the comment?

Reply:
Yes please update the comment, I don't want me or anyone else looking for the bug that doesn't exist :)

Reply:
Done!